content/learn/book/ecs/_index.md

```diff
@@ -0,0 +1,114 @@
+* There is a startup scheduler and a normal scheduler ({{rust_type(type="struct", crate="bevy_ecs", name="Schedule", no_mod=true, plural=false)}})
+* Startup systems run only ONCE before normal systems
+* You can ask normal systems to run once or in an infinite loop
+* You can slow the infinite loop by specifying an additional `k` second sleep delay between each loop
+  - where `k ∈ [0, 9223372036854775807_i64]` seconds (i.e. ~292 billion years)
+* A **System** results from calling the inline {{rust_type(type="trait" crate="bevy_ecs", name="IntoForEachSystem" method="system" no_struct=false)}} method on a `fn`. It is a {{rust_type(type="struct", crate="bevy_ecs", name="SystemFn", no_mod=true, plural=false)}} struct that implements the {{rust_type(type="trait", crate="bevy_ecs", name="System", no_mod=true, plural=false)}} trait and encapsulates the original function 
+* A **System** is always assigned a **stage** in the scheduler
+  - This dictates the group of systems that it runs in parallel with
+  - The default normal stage is `stage::UPDATE`
+  - The default startup stage is `startup_stage::STARTUP`
+* Except when certain data constraints exist (i.e. thread local with exclusive access to {{rust_type(type="struct", crate="bevy_ecs", name="World", no_mod=true, plural=false)}} and {{rust_type(type="struct", crate="bevy_ecs", name="Resources", no_mod=true, plural=false)}})
+  - Startup systems run in no particular order within their **stage** due to parallel execution
+  - Normal systems run in no particular order within their **stage** due to parallel execution
+* Stages exist as an ordered collection
+* For finer control of the order of execution within a scheduler
+  - You can create as many NEW stages in the scheduler as you like
+  - New stages are added immediately *before* or *after* an existing stage (system defined or custom)
+* You can emit and consume custom events
+* You can add new **Components** and **Systems** to a running {{rust_type(type="struct", crate="bevy_app", name="App", no_mod=true, plural=false)}}, i.e. after `.run()`
+  - with `commands:` {{rust_type(type="struct", crate="bevy_ecs", name="Commands", no_mod=true, plural=false)}}
+* {{rust_type(type="struct", crate="bevy_app", name="App", no_mod=true, plural=true)}} also have a mechanism for exiting gracefully
+
+## Create Project
+
+First [create](@/learn/book/getting-started/setup/_index.md#create-a-new-rust-executable-project) a new project called `bevy_ecs_tut` and [add](@/learn/book/getting-started/setup/_index.md#add-bevy-to-your-project-s-cargo-toml) <img src="/assets/bevy_logo_dark.svg" style="height: 1em;margin-bottom:-0.1em" alt="BEVY"/> as a dependency.
+
+If you don't know or remember the current version of Bevy, `cargo` can fetch it for you fast.
+```bash
+$ cargo search bevy
+bevy = "0.1.3"                  # A refreshingly simple data-driven ...
+```
+{{caption(ref=3.1, desc="Truncated output of cargo search")}}
+
+## AppBuilder
```

Comment:
As a general (soft) rule, I think any code section should be preceded by some text explaining whats coming up. Dropping people right into code (especially a big block like this) can be a little overwhelming.

Reply:
It's a good idea. I should shift some of the context given below up above the code.

My experience as a reader of the Rust book at https://doc.rust-lang.org/book, which I otherwise enjoyed, was that it was a bit wordy. I tended to read the code first, and only if I didn't understand what they were trying to say there would I read the preamble.

I'd love a mechanism for finding a balance between the two. Perhaps a very concise breakdown of a paragraph, short-and-sweet, followed by a longer explanation. If I could highlight the breakdown to draw the eye, that would be perfect. I'll experiment with that and see what you think.